iroh-relay/src/client.rs

```diff
@@ -37,6 +36,64 @@ pub(crate) mod streams;
 #[cfg(not(wasm_browser))]
 mod util;
 
+/// Connection errors
+#[allow(missing_docs)]
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum ConnectError {
+    #[error("Invalid URL for websocket {0}")]
+    InvalidWebsocketUrl(Url),
+    #[error("Invalid relay URL {0}")]
+    InvalidRelayUrl(Url),
+    #[error(transparent)]
+    Websocket(#[from] tokio_tungstenite_wasm::Error),
+    #[error(transparent)]
+    Handshake(#[from] HandshakeError),
+    #[error(transparent)]
+    Dial(#[from] DialError),
+    #[error("Unexpected status during upgrade: {0}")]
+    UnexpectedUpgradeStatus(hyper::StatusCode),
+    #[error("Failed to upgrade response")]
+    Upgrade(#[source] hyper::Error),
+    #[error("Invalid TLS servername")]
+    InvalidTlsServername,
+    #[error("No local address available")]
+    NoLocalAddr,
+    #[error("tls connection failed")]
+    Tls(#[source] std::io::Error),
+    #[cfg(wasm_browser)]
+    #[error("The relay protocol is not available in browsers")]
+    RelayProtoNotAvailable,
+}
+
+/// Dialing errors
+#[allow(missing_docs)]
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum DialError {
```

Comment:
Reading just the error types I am rather confused what the difference between `ConnectError` and `DialError` is.  Sounds like the same to me.

It seems `DialError` is a variant of `ConnectError` so maybe it's more of a `ConnectionError`?  But then also `ConnectError` is returned from `Client::connect` so it probably was meant as `ConnectError`  :face_with_spiral_eyes: